lib/extensions/permissions.js

```diff
@@ -102,21 +104,37 @@ async function setAccess (bundleId, permissionsMapping) {
  * @param {string} serviceName - the name of the service. Should be one of
  * `SERVICES` keys.
  * @param {string} simDataRoot - the path to Simulator `data` root
- * @returns {string} - The current status: yes/no/unset
+ * @returns {string} - The current status: yes/no/unset/limited
  * @throws {Error} If there was an error while retrieving permissions.
  */
 async function getAccess (bundleId, serviceName, simDataRoot) {
   const internalServiceName = toInternalServiceName(serviceName);
   const dbPath = path.resolve(simDataRoot, 'Library', 'TCC', 'TCC.db');
-  for (const [sqlValue, status] of [['0', STATUS_NO], ['1', STATUS_YES]]) {
-    const sql = `SELECT count(*) FROM 'access' ` +
-      `WHERE client='${bundleId}' AND allowed=${sqlValue} AND service='${internalServiceName}'`;
-    const count = await execSQLiteQuery(dbPath, sql);
-    if (parseInt(count.split('=')[1], 10) > 0) {
-      return status;
+  const getAccessStatus = async (statusPairs, statusKey) => {
+    for (const [statusValue, status] of statusPairs) {
+      const sql = `SELECT count(*) FROM 'access' ` +
+        `WHERE client='${bundleId}' AND ${statusKey}=${statusValue} AND service='${internalServiceName}'`;
+      const count = await execSQLiteQuery(dbPath, sql);
+      if (parseInt(count.split('=')[1], 10) > 0) {
+        return status;
+      }
     }
+    return STATUS_UNSET;
+  };
+
+  try {
+    // iOS 14 and above
+    return await getAccessStatus(
```

Comment:
shall we instead just explicitly check the xcode version?

Reply:
Yea, potentially. I need to check various environment

Reply:
Perhaps... Xcode 13 can work. Xcode 12 env did not on my local

Reply:
hm, it seems the DB is not Xcode version dependent. iOS 14 created in Xcode 12 had `auth_value`, but did not iOS 12